lib/util/stream_utils.js

```diff
@@ -425,7 +425,44 @@ shaka.util.StreamUtils = class {
     await shaka.util.StreamUtils.getDecodingInfosForVariants(
         manifest.variants, usePersistentLicenses, /* srcEquals= */ false);
     manifest.variants = manifest.variants.filter((variant) => {
+      // See: https://github.com/google/shaka-player/issues/3860
       const video = variant.video;
+      const ContentType = shaka.util.ManifestParserUtils.ContentType;
+      if (video) {
+        let videoCodecs = video.codecs;
+        // For multiplexed streams.
+        if (video.codecs.includes(',')) {
+          const allCodecs = video.codecs.split(',');
+          videoCodecs = shaka.util.ManifestParserUtils.guessCodecs(
+              ContentType.VIDEO, allCodecs);
+          videoCodecs = shaka.util.StreamUtils.patchVp9(videoCodecs);
+          const audioCodecs = shaka.util.ManifestParserUtils.guessCodecs(
+              ContentType.AUDIO, allCodecs);
+          const audioFullType = shaka.util.MimeUtils.getFullOrConvertedType(
+              video.mimeType, audioCodecs, ContentType.AUDIO);
+          if (!MediaSource.isTypeSupported(audioFullType)) {
+            return false;
+          }
+        }
+        videoCodecs = shaka.util.StreamUtils.patchVp9(videoCodecs);
```

Comment:
For multiplexed streams, this was already done in the branch above.  Should this section be in an "else" branch for non-multiplexed only?

Reply:
The If only applies to audio content, the else is not necessary because it is always needed for video, anyway I have seen that something was being done wrong and duplicated.